code/script.js

```diff
@@ -218,108 +221,159 @@ const generateBoard = () => {
+// This function will be invoked when you click on 'Find Out' button.
+// Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+// Then invoke filterCharacters
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+  const { category, value } = currentQuestion;
 
+  if (category === "hair" || category === "eyes") {
+    if (secret[category] === value) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+  } else if (category === "accessories" || category === "other") {
+    if (secret[category].includes(value)) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
   }
-}
+};
 
-// It'll filter the characters array and redraw the game board.
+// This function will filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  const { category, value } = currentQuestion;
+
+  if (category === "accessories") {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+        `Yes, the person wears ${value}! Keep all people that wears ${value}!`
+      );
+      charactersInPlay = charactersInPlay.filter((person) =>
+        person[category].includes(value)
+      );
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+        `No, the person doesn't wear ${value}! Remove all people that wears ${value}!`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => !person[category].includes(value)
+      );
+    }
+  } else if (category === "other") {
+    if (keep) {
+      alert(`Yes, the person is a ${value}! Keep all ${value}s`);
+      charactersInPlay = charactersInPlay.filter((person) =>
+        person[category].includes(value)
+      );
+    } else {
+      alert(`No, the person is not a ${value}! Remove all ${value}s`);
+      charactersInPlay = charactersInPlay.filter(
+        (person) => !person[category].includes(value)
+      );
     }
-  } else if (category === 'other') {
-    // Similar to the one above
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person has ${value} hair! Keep all people with ${value} hair`
```

Comment:
When I played the game I didn't get the right alert pop up when I guess on "eyes" category instead it alerted me about hair category and now I try to see in the code it seems like you don't specify two alerts for "yes" and "no" on eyes category. only on hair which doesn't get specified with "category === hair" like the other and accessories its only stated as the else statement. so It used the hair alerts for both hair and eyes since its the last else conditions. So that might be something you should add if you want :)

Reply:
Wow, thank you Emelie for pointing this out! Will fix asap.